apps/src/javalab/Javalab.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Provider} from 'react-redux';
+import {getStore, registerReducers} from '../redux';
+import JavalabView from './JavalabView';
+import javalab from './javalabRedux';
+import {TestResults} from '@cdo/apps/constants';
+
+/**
+ * On small mobile devices, when in portrait orientation, we show an overlay
+ * image telling the user to rotate their device to landscape mode.  Because
+ * the ailab app is able to render at a minimum width of 480px, we set this
+ * width to be somewhat larger.  We will use this width to set the viewport
+ * on the mobile device, and correspondingly to scale up the overlay image to
+ * properly fit on the mobile device for that viewport.
+ */
+const MOBILE_PORTRAIT_WIDTH = 600;
+
+/**
+ * An instantiable Javalab class
+ */
+
+const Javalab = function() {
+  this.skin = null;
+  this.level = null;
+
+  /** @type {StudioApp} */
+  this.studioApp_ = null;
+};
+
+/**
+ * Inject the studioApp singleton.
+ */
+Javalab.prototype.injectStudioApp = function(studioApp) {
```

Comment:
We haven't been consistent on this everywhere, but this should probably be JavaLab and JavaLab.js (similar to GameLab and SpriteLab). Reasoning: On the product side, we'll always refer to it as two separate words "Java Lab" so camelcasing the two words would make sense.

Reply:
Huh now I'm noticing we're _very_ inconsistent about this.

Reply:
yeah I'm happy to go with either (camel casing probably makes more sense?) but it seemed like keeping lab lowercase was the norm in general

Reply:
Well. haha. When I first read this, I felt strongly that camel casing was the right option. Then I saw what we do everywhere else. I'm happy with whatever you choose.